receiver/prometheusreceiver/metrics_receiver_helper_test.go

```diff
@@ -455,90 +492,54 @@ func compareSummary(count uint64, sum float64, quantiles [][]float64) summaryPoi
 		req := assert.Equal(t, len(quantiles), summaryDataPoint.QuantileValues().Len())
 		if req {
 			for i := 0; i < summaryDataPoint.QuantileValues().Len(); i++ {
-				assert.Equal(t, quantiles[i][0], summaryDataPoint.QuantileValues().At(i).Quantile(), "Summary quantile do not match")
-				assert.Equal(t, quantiles[i][1], summaryDataPoint.QuantileValues().At(i).Value(), "Summary quantile values do not match")
+				assert.Equal(t, quantiles[i][0], summaryDataPoint.QuantileValues().At(i).Quantile(),
+					"Summary quantile do not match")
+				if math.Float64bits(quantiles[i][1]) == value.NormalNaN {
```

Comment:
I thought NaN values were [not comparable](https://go.dev/play/p/pw6aEeYklDk).  You can use `math.IsNaN()` or `math.Float64bits()`, which is the way `value.IsStaleNaN()` works:

```go
math.Float64bits(v) == value.NormalNaN
```

That should only be required if it is necessary to distinguish between the normal and stale NaN values, otherwise `math.IsNaN()` should suffice.

Reply:
@Aneurysm9 thanks. I have updated.